src/generate/apiCollection.test.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import path from "path";
+
+import { expect, default as chai } from "chai";
+import chaiAsPromised from "chai-as-promised";
+
+import ApiCollection from "./apiCollection";
+
+const validRamlFile = path.join(__dirname, "../../test/site.raml");
+const invalidRamlFile = path.join(__dirname, "../../test/search-invalid.raml");
+
+before(() => {
+  chai.use(chaiAsPromised);
+});
+
+describe("Test ApiGroup class read", () => {
+  it("creates an instance from a valid raml file", async () => {
+    const collection = await ApiCollection.read({
+      "Group One": [validRamlFile]
+    });
+    expect(collection["Group One"].apis).to.not.be.empty;
+    expect(collection["Group One"].apis[0].model).to.not.be.empty;
+    expect(collection["Group One"].apis[0].name.original).to.equal("Shop API");
+    expect(collection["Group One"].apis[0].path).to.be.equal(validRamlFile);
+  });
+
+  it("creates an instance from two valid raml files in one group", async () => {
+    const collection = await ApiCollection.read({
+      "Group One": [validRamlFile, validRamlFile]
+    });
+    expect(collection["Group One"].apis.length).to.eq(2);
```

Comment:
```suggestion
    expect(collection["Group One"].apis).to.be.an("array").with.lengthOf(2);
```
Chai provides array-specific assertions.

Reply:
updated